src/utils/hydration.js

```diff
@@ -153,79 +153,28 @@ function injectRepeatedRows({ step, repeatedRowCount }) {
  */
 function calculateDimensions(step) {
     if (step === null) {
-        return { width: null, rows: [], columns: [] };
+        return { definiteWidth: 0, relativeRows: [], relativeColumns: [] };
     }
 
-    const gridTemplateRows = step.grid_rows.map(
-        (row) => `minmax(${row},min-content)`
-    );
-
-    let gridTemplateColumns;
-    if (window.innerWidth >= 768) {
-        gridTemplateColumns = step.grid_columns;
-    } else {
-        const seenColumns = new Set();
-        if (step.progress_bar) {
-            const s = step.progress_bar.column_index;
-            const e = step.progress_bar.column_index_end;
-            for (let i = s; i <= e; i++) {
-                seenColumns.add(i);
-            }
-        }
-
-        step.texts.map((field) => {
-            const s = field.column_index;
-            const e = field.column_index_end;
-            for (let i = s; i <= e; i++) {
-                seenColumns.add(i);
-            }
-        });
-
-        step.buttons.map((field) => {
-            const s = field.column_index;
-            const e = field.column_index_end;
-            for (let i = s; i <= e; i++) {
-                seenColumns.add(i);
-            }
-        });
-
-        step.servar_fields.map((field) => {
-            const s = field.column_index;
-            const e = field.column_index_end;
-            for (let i = s; i <= e; i++) {
-                seenColumns.add(i);
-            }
-        });
-        step.images.map((image) => {
-            const s = image.column_index;
-            const e = image.column_index_end;
-            for (let i = s; i <= e; i++) {
-                seenColumns.add(i);
-            }
-        });
-        gridTemplateColumns = step.grid_columns.map((c, index) =>
-            seenColumns.has(index) ? c : '10px'
-        );
-    }
+    let relativeColumns = step.grid_columns;
```

Comment:
`[...step.grid_columns]` will remove the dangling ref here. Might prevent a bug down the line.